### PR TITLE
Receiving shared pod

### PR DESCRIPTION
**Pull request by IgorShadurin**
This PR added:
-receive information about the pod that has been shared with a user
-add a pod that has been shared with a user to their account

Changed the way the list of pods is displayed. Added 2 getters that return personal pods and added pods separately.

### CODE REVIEW

File: src/pod/utils.ts [hunk 1]
```diff
@@ -137,17 +185,27 @@ export function assertPodName(value: unknown): asserts value is string {
...
+  const allPods =
+    [...pods, ...sharedPods]
+      .map(pod => {
+        if (isPod(pod)) {
+          return `${pod.name},${pod.index}`
+        } else if (isSharedPod(pod)) {
+          return `${pod.name},${bytesToHex(pod.address)}`
+        }
+      })
+      .join('\n') + '\n'
```
Comment (nugaon) [inline src/pod/utils.ts:206]: this mixed data-structure is a bit strange for me, I thought one will be in a different array than the other to clearly have distinction between the two.

File: src/pod/personal-storage.ts [hunk 2]
```diff
@@ -42,37 +51,20 @@ export class PersonalStorage {
...
+
+    const pod = await createPod(
       this.accountData.connection.bee,
-      prepareEthAddress(this.accountData.wallet!.address),
-      this.accountData.connection.options?.downloadOptions,
+      this.accountData.connection,
+      this.accountData.wallet!,
+      {
+        name,
+        index: 0,
```
Comment (nugaon) [inline src/pod/personal-storage.ts:61]: is the index always 0?

**Review by nugaon — COMMENTED** [review]
LGTM overall, but I don't see the _utilization_ of the shared pods such as listing directories of a shared pod, etc.

I had some comments regarding to the data-structure being used for sharing pods, but those are not as important because later we will change the used metadata-structure format anyway.

File: src/pod/utils.ts (hunk 1 above)
Comment (IgorShadurin) [reply src/pod/utils.ts:206]: unfortunately, such a structure in fairos. We need to get away from this.

**Comment by IgorShadurin** [comment]
@nugaon about utilization. Now method `fdp.personalStorage.list()` returns `List` object which can return not only simple pods with methods `getPods()`, but also shared pods with `getSharedPods()`

File: src/pod/personal-storage.ts (hunk 2 above)
Comment (IgorShadurin) [reply src/pod/personal-storage.ts:61]: no, later the index calculates with correct value

**Comment by nugaon** [comment]
> @nugaon about utilization. Now method `fdp.personalStorage.list()` returns `List` object which can return not only simple pods with methods `getPods()`, but also shared pods with `getSharedPods()`

yes, but what can you do with the object that came from the `getSharedPods()`?

**Pull request merged**